server/plugin/plugin.go

```diff
@@ -41,6 +44,14 @@ func (p *Plugin) OnActivate() error {
 	}
 
 	p.router = p.initializeAPI()
+	p.dmChannelCache = gcache.New(p.getConfiguration().ChannelCacheSize).ARC().Build()
```

Comment:
What are the reasons an ARC was picked over the alternatives?

Reply:
Because it is better in performance than both LRU and LFU. https://en.wikipedia.org/wiki/Adaptive_replacement_cache

Reply:
"Performance" basically means reducing the number of "cache miss" and avoiding the RPC call a maximum no. of times. In our use case, several users will be chatting at the same time and our cache is storing all channel IDs whether it's a DM channel or not. Therefore, this hook `MessageHasBeenPosted` will be called many times and the cache will be checked every time. So, I think we should choose the most optimal algorithm based on performance.
What are your thoughts on this?

Reply:
Optimising for cache missing seems fine to me :+1: